# Exploratory Data Analysis Notebook

## Contents

- Import Libaries
- Data & Features
- Exploratory Data Analysis (EDA)
    - Funtions
    - Shape & Size
    - Information & Statistical Summary
    - Unique & Missing Values
    - Categoriacal & Numerical Variables
    - Tableue Visualization
    - Class imbalance
    - Correlation
    - Outliers

## Import Libaries

In [2]:
import numpy as np 
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns
import os

In [2]:
sns.set(style="darkgrid")

## Data & Features

In [3]:
path = "../data/bank+marketing/bank-additional/bank-additional/bank-additional-full.csv"
df = pd.read_csv(path, sep=';')

In [4]:
df.head()

age        job  marital    education  default housing loan    contact  \
0   56  housemaid  married     basic.4y       no      no   no  telephone   
1   57   services  married  high.school  unknown      no   no  telephone   
2   37   services  married  high.school       no     yes   no  telephone   
3   40     admin.  married     basic.6y       no      no   no  telephone   
4   56   services  married  high.school       no      no  yes  telephone   

  month day_of_week  ...  campaign  pdays  previous     poutcome emp.var.rate  \
0   may         mon  ...         1    999         0  nonexistent          1.1   
1   may         mon  ...         1    999         0  nonexistent          1.1   
2   may         mon  ...         1    999         0  nonexistent          1.1   
3   may         mon  ...         1    999         0  nonexistent          1.1   
4   may         mon  ...         1    999         0  nonexistent          1.1   

   cons.price.idx  cons.conf.idx  euribor3m  nr.employed   y  
0          93.994          -36.4      4.857       5191.0  no  
1          93.994          -36.4      4.857       5191.0  no  
2          93.994          -36.4      4.857       5191.0  no  
3          93.994          -36.4      4.857       5191.0  no  
4          93.994          -36.4      4.857       5191.0  no  

[5 rows x 21 columns]

## Description of Features of the Dataset

- age : (numeric)

- job : type of job (categorical: 'admin.','blue collar','entrepreneur','housemaid','management','retired','self employed','services','student','technician','unemployed','unknown')

- marital : marital status (categorical: 'divorced','married','single','unknown'; note: 'divorced' means divorced or widowed)

- education (categorical): 'basic.4y','basic.6y','basic.9y','high.school','illiterate','professional.course','university.degree','unknown'

- default: has credit in default? (categorical: 'no','yes','unknown')

- housing: has housing loan? (categorical: 'no','yes','unknown')

- loan: has personal loan? (categorical: 'no','yes','unknown')

- related with the last contact of the current campaign:

- contact: contact communication type (categorical: 'cellular','telephone')

- month: last contact month of year (categorical: 'jan', 'feb', 'mar', ..., 'nov', 'dec')

- day_of_week: last contact day of the week (categorical: 'mon','tue','wed','thu','fri')

- duration: last contact duration, in seconds (numeric). Important note: this attribute highly affects the output target (e.g., if duration=0 then y='no'). Yet, the duration is not known before a call is performed. Also, after the end of the call y is obviously known. Thus, this input should only be included for benchmark purposes and should be discarded if the intention is to have a realistic predictive model.

### other attributes:

- campaign: number of contacts performed during this campaign and for this client (numeric, includes last contact)

- pdays: number of days that passed by after the client was last contacted from a previous campaign (numeric; 999 means client was not previously contacted)

- previous: number of contacts performed before this campaign and for this client (numeric)

- poutcome: outcome of the previous marketing campaign (categorical: 'failure','nonexistent','success')

### social and economic context attributes

- emp.var.rate: employment variation rate - quarterly indicator (numeric)

- cons.price.idx: consumer price index - monthly indicator (numeric)

- cons.conf.idx: consumer confidence index - monthly indicator (numeric)

- euribor3m: euribor 3 month rate - daily indicator (numeric)

- nr.employed: number of employees - quarterly indicator (numeric)

### Output variable (desired target):

y - has the client subscribed to a term deposit? (binary: 'yes','no')

## Exploratory Data Analysis


### Shape and Size

In [3]:
print(data.name,'shape:',data.shape)
print(data.name,'size:',data.size)

NameError: name 'data' is not defined

### Information and Statistical summary

In [ ]:
print(data.info())

In [ ]:
dataset.describe().T


In [ ]:

cols = data.columns
for i in cols:
    if data[i].dtype == 'O':
        print('Unique values in',i,'are',data[i].unique())
        print('----------------------------------------------')

In [ ]:
print(data.isnull().sum())

In [ ]:
for p in figure.patches:
    figure.annotate(format(p.get_height()), (p.get_x() + p.get_width() / 2.,
                        p.get_height()), ha = 'center', va = 'center', 
                        xytext = (0, 10), textcoords = 'offset points')